src/status_ctrl.js

```diff
@@ -317,24 +318,19 @@ export class StatusPluginCtrl extends MetricsPanelCtrl {
 
 	handleMaxAlertsToShow() {
 		if(this.panel.maxAlertNumber != null && this.panel.maxAlertNumber >= 0) {
-			let currentAlertsNumber = this.panel.maxAlertNumber;
-			if(this.disabled.length > currentAlertsNumber) {
-				this.disabled = this.disabled.slice(0,currentAlertsNumber);
+			let currentMaxAllowedAlerts = this.panel.maxAlertNumber;
+			let filteredOutAlerts = 0;
+			let arrayNamesToSlice = ["disabled", "crit", "warn", "display"];
+			for(let i=0; i<arrayNamesToSlice.length; i++) {
```

Comment:
Also, we could put all arrays under a single object, and loose the string array altogether, but that is optional.